src/WindowsTracing/KrabsTracer.cpp

```diff
@@ -103,38 +108,51 @@ void KrabsTracer::SetupStackTracing() {
                                      (void*)&interval, sizeof(TRACE_PROFILE_INTERVAL));
   ORBIT_CHECK(status == ERROR_SUCCESS);
 
-  // Initialize ETW stack tracing. Note that this must be executed after trace_.open() as
+  // Initialize ETW stack tracing. Note that this must be executed after kernel_trace_.open() as
   // set_trace_information needs a valid session handle.
   CLASSIC_EVENT_ID event_id = {0};
   event_id.EventGuid = krabs::guids::perf_info;
   event_id.Type = kSampledProfileEventSampleProfile;
-  trace_.set_trace_information(TraceStackTracingInfo, &event_id, sizeof(CLASSIC_EVENT_ID));
+  kernel_trace_.set_trace_information(TraceStackTracingInfo, &event_id, sizeof(CLASSIC_EVENT_ID));
 }
 
 void KrabsTracer::Start() {
-  ORBIT_CHECK(trace_thread_ == nullptr);
+  ORBIT_CHECK(kernel_trace_thread_ == nullptr);
+  ORBIT_CHECK(user_trace_thread_ == nullptr);
   context_switch_manager_ = std::make_unique<ContextSwitchManager>(listener_);
   SetIsSystemProfilePrivilegeEnabled(true);
-  log_file_ = trace_.open();
+  log_file_ = kernel_trace_.open();
   if (IsProviderEnabled(ProviderFlags::kStackWalk)) {
     SetupStackTracing();
   }
-  trace_thread_ = std::make_unique<std::thread>(&KrabsTracer::Run, this);
+  kernel_trace_thread_ = std::make_unique<std::thread>(&KrabsTracer::KernelTraceThread, this);
+  user_trace_thread_ = std::make_unique<std::thread>(&KrabsTracer::UserTraceThread, this);
 }
 
 void KrabsTracer::Stop() {
-  ORBIT_CHECK(trace_thread_ != nullptr && trace_thread_->joinable());
-  trace_.stop();
-  trace_thread_->join();
-  trace_thread_ = nullptr;
+  ORBIT_CHECK(kernel_trace_thread_ != nullptr && kernel_trace_thread_->joinable());
+  kernel_trace_.stop();
+  kernel_trace_thread_->join();
+  kernel_trace_thread_ = nullptr;
+
+  ORBIT_CHECK(user_trace_thread_ != nullptr && user_trace_thread_->joinable());
+  user_trace_.stop();
+  user_trace_thread_->join();
+  user_trace_thread_ = nullptr;
```

Comment:
We need a function here.

Reply:
It would have to be a templated function... I don't think it adds much, I did put the two blocks in their respective function however.